machine-check/README.md

```diff
@@ -1,3 +1,123 @@
+
+- `(Closed) --[open@Control<Opening>]--> (Opening)`
+- `(Opening) --[update@Door<Opening>]--> (Opening)`
+- `(Opening) --[open@Door<Opened>]--> (Open)`
+- `(Open) --[close@Control<Closing>]--> (Closing)`
+- `(Closing) --[update@Door<Closing>]--> (Closing)`
+- `(Closing) --[close@Door<Closed>]--> (Closed)`
+
+The Control can initiate opening and closing while the Door provides progress updates and states when each movement has been completed.
+
+## How to use this library
+
+This library is typically used within your unit tests to check the structure of the machines you’ve written.
+For this we need to provide two pieces: the desired swarm protocol and the event subscriptions of your machine roles.
+Continuing the example above, it could look like this:
+
+```ts
+import { Door, Control, HangarBay } from './example-proto.js'
+import { SwarmProtocolType, checkProjection, checkSwarmProtocol } from '@actyx/machine-check'
+
+const swarmProtocol: SwarmProtocolType = {
+  initial: 'Closed',
+  transitions: [
+    {
+      source: 'Closed',
+      target: 'Opening',
+      label: { cmd: 'open', role: 'Control', logType: ['opening'] },
+    },
+    {
+      source: 'Opening',
+      target: 'Opening',
+      label: { cmd: 'update', role: 'Door', logType: ['opening'] },
+    },
+    {
+      source: 'Opening',
+      target: 'Open',
+      label: { cmd: 'open', role: 'Door', logType: ['opened'] },
+    },
+    {
+      source: 'Open',
+      target: 'Closing',
+      label: { cmd: 'close', role: 'Control', logType: ['closing'] },
+    },
+    {
+      source: 'Closing',
+      target: 'Closing',
+      label: { cmd: 'update', role: 'Door', logType: ['closing'] },
+    },
+    {
+      source: 'Closing',
+      target: 'Closed',
+      label: { cmd: 'close', role: 'Door', logType: ['closed'] },
+    },
+  ],
+}
+
+const subscriptions = {
+  Control: ['closing', 'closed', 'opening', 'opened'],
+  Door: ['closing', 'closed', 'opening', 'opened'],
+}
+
+console.log(checkSwarmProtocol(swarmProtocol, subscriptions))
+console.log(
+  checkProjection(
+    swarmProtocol,
+    subscriptions,
+    'Control',
+    Control.Control.createJSONForAnalysis(Control.Closed),
+  ),
+)
+console.log(
+  checkProjection(
+    swarmProtocol,
+    subscriptions,
+    'Door',
+    Door.Door.createJSONForAnalysis(Door.Closed),
+  ),
+)
+```
+
+Instead of `console.log()` you’d normally assert that the result is `{"type":"OK"}`.
+In the example as given you’ll instead be notified that the overall protocol has a flaw:
+
+```text
+{
+  type: 'ERROR',
+  errors: [
+    'guard event type opening appears in transitions from multiple states',
+    'guard event type closing appears in transitions from multiple states'
+  ]
+}
+```
+
+This means that our clever reuse of the `opening` and `closing` event types for dual purposes may not be so clever after all — the `update` commands should yield more specific `openingProgress` and `closingProgress` event types instead.
```

Comment:
If the user doesn't read the example (which is in another document) this sentence doesn't make much sense. 
I'm reviewing in GH's file order and I got confused by "clever reuse".
Adding a "you should read X before this" or re-working this explanation to include a more in-depth description of the issue